From 6bca9742c8667ed956ac94a1d9055509b7b3c24e Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Fri, 15 Jun 2018 17:06:10 +0800
Subject: [PATCH] input: touchscreen: gt1x: Fix NULL pointer exception in
 unbind

We're using devm_regulator_get, no need to put it manually.

Change-Id: Ida960d938d7557d0174c53e952d80161650ccf1a
Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 drivers/input/touchscreen/gt1x/gt1x.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/drivers/input/touchscreen/gt1x/gt1x.c b/drivers/input/touchscreen/gt1x/gt1x.c
index f0cfa078b328..75896a9e6215 100644
--- a/drivers/input/touchscreen/gt1x/gt1x.c
+++ b/drivers/input/touchscreen/gt1x/gt1x.c
@@ -357,11 +357,6 @@ static void gt1x_remove_gpio_and_power(void)
 	if (gpio_is_valid(gt1x_rst_gpio))
 		gpio_free(gt1x_rst_gpio);
 
-#ifdef GTP_CONFIG_OF
-	if (vdd_ana)
-		regulator_put(vdd_ana);
-#endif
-
 	if (gt1x_i2c_client && gt1x_i2c_client->irq)
 		free_irq(gt1x_i2c_client->irq, gt1x_i2c_client);
 }
-- 
2.35.3

